File: br/football-br.js
let TIMES = {
    timeA: {nome: 'Brasil', pontos: 0, gols_feitos: 0, gols_diferenca: 0, gols_sofridos: 0,},
    timeB: {nome: 'França', pontos: 0, gols_feitos: 0, gols_diferenca: 0, gols_sofridos: 0,},
    timeC: {nome: 'Israel', pontos: 0, gols_feitos: 0, gols_diferenca: 0, gols_sofridos: 0,},
    timeD: {nome: 'Angola', pontos: 0, gols_feitos: 0, gols_diferenca: 0, gols_sofridos: 0,},
}
let PARTIDAS = [
  // Turno 1
    // Rodada 1
    {casa: TIMES.timeA, golsCasa: 0, fora: TIMES.timeB, golsFora: 0},
    {casa: TIMES.timeC, golsCasa: 0, fora: TIMES.timeD, golsFora: 0},
    // Rodada 2
    {casa: TIMES.timeA, golsCasa: 0, fora: TIMES.timeC, golsFora: 0},
    {casa: TIMES.timeB, golsCasa: 0, fora: TIMES.timeD, golsFora: 0},
    // Rodada 3
    {casa: TIMES.timeA, golsCasa: 0, fora: TIMES.timeD, golsFora: 0},
    {casa: TIMES.timeC, golsCasa: 0, fora: TIMES.timeB, golsFora: 0},
  // Turno 2
    // Rodada 4
    {fora: TIMES.timeA, golsCasa: 0, casa: TIMES.timeB, golsFora: 0},
    {fora: TIMES.timeC, golsCasa: 0, casa: TIMES.timeD, golsFora: 0},
    // Rodada 5
    {fora: TIMES.timeA, golsCasa: 0, casa: TIMES.timeC, golsFora: 0},
    {fora: TIMES.timeB, golsCasa: 0, casa: TIMES.timeD, golsFora: 0},
    // Rodada 6
    {fora: TIMES.timeA, golsCasa: 0, casa: TIMES.timeD, golsFora: 0},
    {fora: TIMES.timeB, golsCasa: 0, casa: TIMES.timeC, golsFora: 0},
]
function criarGols(max) {
    return Math.floor(Math.random() * max)
}
function distribuirGols(partida) {
    partida.golsCasa = criarGols(4)
    partida.golsFora = criarGols(4)
}
function aplicarGolsPartidas(partidas) {
    for (let partida of partidas) {
      distribuirGols(partida)
    }
}
function mostrarPartidas(partida) {
    console.log(`${partida.casa.nome}    ${partida.golsCasa}  x  ${partida.golsFora}   ${partida.fora.nome}`)
}
function relatorioPartidas(partidas) {
    let numeroPartida = 1
    let numeroRodada = 1
    let numeroTurno = 1
  
    for (let partida of partidas) {
      if (numeroPartida % 6 === 1) {
        console.log(`\nTurno ${numeroTurno}`)
        numeroTurno++
      }
      if (numeroPartida % 2 === 1) {
        console.log(`\nRodada ${numeroRodada}`)
        numeroRodada++
      }
      mostrarPartidas(partida)
      numeroPartida++
    }
}
function somarPontos(partidas) {
    for (let {casa, fora, golsCasa, golsFora} of partidas) {
      if (golsCasa > golsFora) {
        casa.pontos += 3
      } else if (golsCasa < golsFora) {
        fora.pontos += 3
      } else {
        casa.pontos++
        fora.pontos++
      }
      casa.gols_feitos += golsCasa
      fora.gols_feitos += golsFora
  
      casa.gols_sofridos += golsFora
      fora.gols_sofridos += golsCasa
  
      casa.gols_diferenca += golsCasa - golsFora
      fora.gols_diferenca += golsFora - golsCasa
    }
}
function classificarTimes(times) {
    let listaTimes = Object.values(times)
  
    listaTimes.sort((time1, time2) => {
      if(time1.pontos !== time2.pontos) {
        return time2.pontos - time1.pontos
      }
      if(time2.gols_feitos !== time2.gols_feitos) {
        return time2.gols_feitos - time1.gols_feitos
      }
      if(time1.gols_diferenca !== time2.gols_diferenca) {
        return time2.gols_diferenca - time1.gols_diferenca
      }
      return time1.gols_sofridos - time2.gols_sofridos
    })
    return listaTimes
}
function final(primeiro, segundo) {
    let partida = {fora: primeiro, golsCasa: 0, casa: segundo, golsFora: 0}
  
    distribuirGols(partida)
    mostrarPartidas(partida)
  
    let campeao = null
  
    if(partida.golsCasa === partida.golsFora) {
      let penaltiCasa = criarGols(6)
      let penaltiFora = criarGols(6)
  
      while(penaltiCasa === penaltiFora) {
        penaltiCasa += criarGols(2)
        penaltiFora += criarGols(2)
      }
  
      console.log('\nPênaltis')
      console.log(`${partida.casa.nome}     ${penaltiCasa}  x  ${penaltiFora}     ${partida.fora.nome}`)
  
      if(penaltiCasa > penaltiFora) {
        campeao = partida.casa
      } else {
        campeao = partida.fora
      }
    } else {
      if(partida.golsCasa > partida.golsFora) {
        campeao = partida.casa
      } else {
        campeao = partida.fora
      }    
    }
    console.log(`Campeão ${campeao.nome}`)
}  
aplicarGolsPartidas(PARTIDAS)
relatorioPartidas(PARTIDAS)
somarPontos(PARTIDAS)
let classificacao = classificarTimes(TIMES)
console.table(classificacao)

let [primeiro, segundo] = classificacao
final(primeiro, segundo)

File: br/football-br.py
from random import randint 
TIMES = {
    'time_a': {'nome': "Brasil",'pontos': 0,'gols_feitos': 0,'gols_diferenca': 0,'gols_sofridos': 0},
    'time_b': {'nome': "França",'pontos': 0,'gols_feitos': 0,'gols_diferenca': 0,'gols_sofridos': 0},
    'time_c': {'nome': "Israel",'pontos': 0,'gols_feitos': 0,'gols_diferenca': 0,'gols_sofridos': 0},
    'time_d': {'nome': "Angola",'pontos': 0,'gols_feitos': 0,'gols_diferenca': 0,'gols_sofridos': 0},
}

PARIDAS = [
# Turno 1
    # Rodada 1
    {'casa': TIMES['time_a'], 'gols_casa': 0, 'fora': TIMES['time_b'], 'gols_fora': 0},
    {'casa': TIMES['time_c'], 'gols_casa': 0, 'fora': TIMES['time_d'], 'gols_fora': 0},
    # Rodada 2
    {'casa': TIMES['time_a'], 'gols_casa': 0, 'fora': TIMES['time_c'], 'gols_fora': 0},
    {'casa': TIMES['time_b'], 'gols_casa': 0, 'fora': TIMES['time_d'], 'gols_fora': 0},
    # Rodada 3
    {'casa': TIMES['time_a'], 'gols_casa': 0, 'fora': TIMES['time_d'], 'gols_fora': 0},
    {'casa': TIMES['time_c'], 'gols_casa': 0, 'fora': TIMES['time_b'], 'gols_fora': 0},
# Turno 2
    # Rodada 4
    {'casa': TIMES['time_a'], 'gols_casa': 0, 'fora': TIMES['time_b'], 'gols_fora': 0},
    {'casa': TIMES['time_c'], 'gols_casa': 0, 'fora': TIMES['time_d'], 'gols_fora': 0},
    # Rodada 5
    {'casa': TIMES['time_a'], 'gols_casa': 0, 'fora': TIMES['time_c'], 'gols_fora': 0},
    {'casa': TIMES['time_b'], 'gols_casa': 0, 'fora': TIMES['time_d'], 'gols_fora': 0},
    # Rodada 6
    {'casa': TIMES['time_a'], 'gols_casa': 0, 'fora': TIMES['time_d'], 'gols_fora': 0},
    {'casa': TIMES['time_c'], 'gols_casa': 0, 'fora': TIMES['time_b'], 'gols_fora': 0},
]
def criar_gols():
    return randint(0,3) 
def distribuir_gols(partida):
    partida['gols_casa'] = criar_gols()
    partida['gols_fora'] = criar_gols()
def aplicar_gols_partidas(partidas):
    for partida in partidas:
        distribuir_gols(partida)
def mostrar_partida(partida):
    print(f"{partida['casa']['nome']} {partida['gols_casa']} X {partida['gols_fora']} {partida['fora']['nome']}")
def relatorio_partidas(partidas):
    numero_partida = 1
    numero_rodada = 1
    numero_turno = 1

    for partida in partidas:
        if (numero_partida % 6) == 1:
            print(f"\nTurno {numero_turno}")
            numero_turno += 1
        if (numero_partida % 2) == 1:
            print(f"\nRodada {numero_rodada}")
            numero_rodada += 1
        mostrar_partida(partida)
        numero_partida += 1
def somar_pontos(partidas):
    for partida in partidas:
        if partida['gols_casa'] > partida['gols_fora']:
            partida['casa']['pontos'] += 3
        elif partida['gols_casa'] < partida['gols_fora']:
            partida['fora']['pontos'] += 3
        else:
            partida['casa']['pontos'] += 1
            partida['fora']['pontos'] += 1

        partida['casa']['gols_feitos'] += partida['gols_casa']
        partida['fora']['gols_feitos'] += partida['gols_fora']

        partida['casa']['gols_diferenca'] += partida['gols_casa'] - partida['gols_fora']
        partida['fora']['gols_diferenca'] += partida['gols_fora'] - partida['gols_casa']

        partida['casa']['gols_sofridos'] += partida['gols_fora']
        partida['fora']['gols_sofridos'] += partida['gols_casa']
def classificar_times():
    lista_times = list(TIMES.values()) 
    lista_times = sorted(lista_times, key = lambda time: (time['pontos'], time['gols_feitos'], time['gols_diferenca'], time['gols_sofridos']), reverse = True)
    print("\n", sorted(lista_times, key = lambda time: (time['pontos'], time['gols_feitos'], time['gols_diferenca'], time['gols_sofridos']), reverse = True))
    return lista_times
def final(primeiro, segundo):
    partida = {'casa': primeiro, 'gols_casa': 0, 'fora': segundo, 'gols_fora': 0}
    
    distribuir_gols(partida)
    print("\n")
    mostrar_partida(partida)

    campeao = None

    if partida['gols_casa'] == partida['gols_fora']:
        penalti_casa = randint(0,5)
        penalti_fora = randint(0,5)
    
        while penalti_casa == penalti_fora:
            penalti_casa += randint(0,2)
            penalti_fora += randint(0,2)

        print("\nPênaltis")
        print(f"{partida['casa']['nome']} {penalti_casa} x {penalti_fora} {partida['fora']['nome']}")

        if penalti_casa > penalti_fora:
            campeao = partida['casa']
        else:
            campeao = partida['fora']
    else:
        if partida['gols_casa'] > partida['gols_fora']:
            campeao = partida['casa']
        else:
            campeao = partida['fora']
    print(f"Campeão {campeao['nome']}")
aplicar_gols_partidas(PARIDAS)
relatorio_partidas(PARIDAS)
somar_pontos(PARIDAS)

classificacao = classificar_times()
primeiro, segundo = classificacao[0], classificacao[1]
final(primeiro, segundo)
